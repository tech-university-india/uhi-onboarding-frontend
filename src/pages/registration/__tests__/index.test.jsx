import Registration from ".."
import { fireEvent, render, screen } from "@testing-library/react";

describe("RegisterForm", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Registration />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the correct text", () => {
    render(<Registration />);
    expect(screen.getByText("Register")).toBeInTheDocument();
  })

  it("should render aadhar otp when clicked on Register button", () => {
    render(<Registration />);
    const registerButton = screen.getByText("Register");
    fireEvent.click(registerButton);
    expect(screen.getByText("Submit")).toBeInTheDocument();
  })

});
