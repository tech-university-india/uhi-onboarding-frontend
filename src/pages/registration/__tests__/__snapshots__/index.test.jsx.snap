// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should render correctly 1`] = `
<DocumentFragment>
  <header
    class="bg-primary shadow-header p-3 w-full rounded-md rounded-t-none"
  >
    <h1
      class="text-3xl font-light"
    >
      Ayush
    </h1>
  </header>
  <div
    class="p-7"
  >
    <h1
      class="mb-10 text-xl"
    >
      Registration
    </h1>
    <p
      data-testid="handle-aadhar"
    />
    <input
      class="disabled:text-preFillText p-3 w-full rounded-lg shadow-textBox bg-textBox"
      placeholder="Aadhar Number"
      type="text"
      value=""
    />
    <button
      class="bg-submit mt-10 block mx-auto my-0 shadow-xl px-8 py-3 rounded-xl text-base font-medium "
    >
      Register
    </button>
    <div
      class="inline-flex mt-10"
    >
      <p
        class="text-md"
      >
        Already have an ABHA number? 
      </p>
      <button
        class="bg-submit shadow-xl px-8 py-3 rounded-xl text-base font-medium "
      >
        Login
      </button>
    </div>
  </div>
</DocumentFragment>
`;
